### PR TITLE
ADRs to cover recent domain and repo agreements

### PR DESCRIPTION
**Pull request by jackgoodby**
# Description

[Describe what this change achieves]

# Issues Resolved

[List any existing issues this PR resolves]

# Contribution Checklist

- [ ] Terraform plans
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: docs/architecture/decisions/0009-api-domain-structure.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+#### [api-domain]
+
+* api.sirius.opg.digital
+
+#### examples
+
+* https://dev.api.sirius.opg.digital/health
+* https://preprod.api.sirius.opg.digital/health
+* https://api.sirius.opg.digital/health
```
Comment (sixdaysandy) [inline docs/architecture/decisions/0009-api-domain-structure.md]: The none production URL's don't match the standard that Sirius uses (or is about to use).
`https://api.dev.sirius.opg.digital
https://api.preproduction.sirius.opg.digital
https://api.sirius.opg.digital`

Comment (mattmachell) [reply docs/architecture/decisions/0009-api-domain-structure.md]: So worth updating this to include the new domains that Andy and Jim have sorted for us.

api.opg.service.justice.gov.uk
deputy-reporting.api.opg.service.justice.gov.uk
pre.deputy-reporting.api.opg.service.justice.gov.uk
dev.deputy-reporting.api.opg.service.justice.gov.uk
pr-1234.dev.deputy-reporting.api.opg.service.justice.gov.uk

File: docs/architecture/decisions/0009-api-domain-structure.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+We need to establish a domain structure which
+
+* Is product agnostic
+* Is consistent across the opg-data service
+
+## Decision
+
+We will adopt the pattern:
+
+[account-type].[api-domain]/[restapi-name]
```
Comment (mattmachell) [inline docs/architecture/decisions/0009-api-domain-structure.md]: So pattern agreed with ops is:

[pull-request-id].[account-stage].[microservice-domain].api.opg.service.justive.gov.uk 

Where [pull-request-id] is for ephemeral dev environments.
Where [account-stage] is the stage in our path to live/accounts i.e. dev,pre and empty for prod services.
Where [microservice-domain] is the scoped domain of the microservice/integration in in question.

File: docs/architecture/decisions/0010-api-repo-structure.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+Date: 2020-02-03
+
+## Status
+
+Proposed
+
+## Context
+
+AWS infrastructure such as API gateway endpoints and lambdas are deployed via terraform, and so are represented at a filesystem-level by terraform .tf files
+
```
Comment (mattmachell) [inline docs/architecture/decisions/0010-api-repo-structure.md]: Worth being explicit on the form of the repo naming convention. i.e.

`opg-data-[microservice-domain]`

File: docs/architecture/decisions/0010-api-repo-structure.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+## Decision
+
+Repo naming convention:
+opg-data-[microservice-domain]
+
+* .tf files pertaining to API Gateway infra to be stored in the [opg-sirius-api-gateway repo](https://github.com/ministryofjustice/opg-sirius-api-gateway)
```
Comment (sixdaysandy) [inline docs/architecture/decisions/0010-api-repo-structure.md]: API Gateway, as a piece of infrastructure currently lives in the opg-sirius-infrastructure repo, as the decision was made that for integrations into Sirius this was the logical place to store it.
opg-sirius-api-gateway repo needs migrating to our new pattern and deleting.

Comment (mattmachell) [reply docs/architecture/decisions/0010-api-repo-structure.md]: Worth adding that as a consequence of the ADR.

**Pull request merged**